src/app/components/sideNav/sidenav.controller.js

```diff
@@ -59,7 +59,7 @@
       description: 'Previous of health recommendations',
       callback: function() {
         var i = states.indexOf($state.current.name);
-        var previousState = states[(i-1)%4];
+        var previousState = states[(i+3)%4];
```

Comment:
$state.go(previousState); ---> jump to previous state.
Line 61. var i = number of the current state
62. Previous state is current  number + 3 % 4. For example, 1 (sleep state) + 3 % 4= 0 (nutrition state)